### DISCUSSION TITLE
Improvement: Optional librechat.yaml path via environment variable

### DISCUSSION THREAD

**Discussion by ndrsfel** [danny-avila/LibreChat#1825] [Feature Requests & Suggestions]
Hey all, I have LibreChat running in an Azure Web App via the default Docker container deployment. Whereas path mapping to folders is possible out of the box, overwriting a file is not. This limits me in configuring LC with the standard Docker container. 

I‘d therefore suggest to introduce a new environment variable `CONFIG_PATH` to change the location of `librechat.yaml`. For example 

```javascript
const fs = require('fs');
const path = require('path');

// Assuming projectRoot is defined elsewhere
const defaultConfigPath = path.resolve(projectRoot, 'librechat.yaml');

let i = 0;

async function loadCustomConfig() {
  // Use CONFIG_PATH if set, otherwise fallback to defaultConfigPath
  const configPath = process.env.CONFIG_PATH || defaultConfigPath;

  // Attempt to load the YAML file
  const customConfig = loadYaml(configPath);
  if (!customConfig) {
    if (i === 0) {
      logger.info(
        'Custom config file missing or YAML format invalid.\n\nCheck out the latest config file guide for configurable options and features.\nhttps://docs.librechat.ai/install/configuration/custom_config.html\n\n',
      );
      i++;
    }
    return null;
  }

  // Continue with customConfig processing
  return customConfig;
}
```

Let me know what you think. Happy to create a PR for this after discussions.

**Comment by fuegovic** [comment]
I personally like this, it could be very useful in some situation

**Comment by danny-avila** [comment]
This is great! Feel free to submit a PR or I will implement it otherwise.

**Comment by ndrsfel** [comment]
Solved by #1858
